methods/basesim.R

```diff
@@ -0,0 +1,205 @@
+
+## tpt sims
+
+# run sims  
+
+sims$tpt <- model$tpt %>% 
+  escalation::simulate_trials(next_dose = start_dose, n_sims$tpt, true_dlt_ss, NULL)
+
+# find selection probs  
+
+selection$tpt <- sims$tpt %>% 
+  escalation::prob_recommend()
+
+# find no. treated at dose
+
+treated$tpt <- colSums(sims$tpt %>% 
+  escalation::n_at_dose())
+
+treatedpct$tpt <- treated$tpt / sum(treated$tpt)
+
+# coerce into selection table
+
+selection_df$tpt <- data.frame(selection$tpt)
+
+selection_tab$tpt <- rbind(t(selection_df$tpt), c(NA,true_dlt_ss))
+
+rownames(selection_tab$tpt) <- c("Dose Selected by Model", "True Toxicity Probabilities")
+
+# coerce into treatment table
+
+treatedpct_df$tpt <- data.frame(treatedpct$tpt)
+
+treatment_tab$tpt <- rbind(t(treatedpct_df$tpt),true_dlt_ss)
+
+rownames(treatment_tab$tpt) <- c("Patients Treated by Model", "True Toxicity Probabilities")
+
+# i) accuracy
+
+dist_accuracy$tpt <- recommended_dose(sims$tpt)
+mean_accuracy$tpt <- length(subset(dist_accuracy$tpt,dist_accuracy$tpt == best_dose_level)) / n_sims$tpt
+#hist(dist_accuracy$tpt,breaks=10)
+
+# ii) risk of overdosing
+
+dist_overdose$tpt <- num_tox(sims$tpt)
+mean_overdose$tpt <- mean(dist_overdose$tpt)
+#hist(dist_overdose$tpt,breaks=10)
+
+# iii) trial length
+
+dist_length$tpt <- sims$tpt %>% escalation::trial_duration()
+mean_length$tpt <- mean(dist_length$tpt)
+#hist(dist_length$tpt,breaks=10)
+
+#############################################################################################################################################################
+
+## crm sims
+
+# run sims  
+
+sims$crm <- model$crm %>% 
+  escalation::simulate_trials(next_dose = start_dose, n_sims$crm, true_dlt_ss, NULL)
+
+# find selection probs  
+
+selection$crm <- sims$crm %>% 
+  escalation::prob_recommend()
+
+# find no. treated at dose
+
+treated$crm <- colSums(sims$crm %>% 
+  escalation::n_at_dose())
+
+treatedpct$crm <- treated$crm / sum(treated$crm)
+
+# coerce into selection table
+
+selection_df$crm <- data.frame(selection$crm)
+
+selection_tab$crm <- rbind(t(selection_df$crm), c(NA,true_dlt_ss))
+
+rownames(selection_tab$crm) <- c("Dose Selected by Model", "True Toxicity Probabilities")
+
+# coerce into treatment table
+
+treatedpct_df$crm <- data.frame(treatedpct$crm)
+
+treatment_tab$crm <- rbind(t(treatedpct_df$crm),true_dlt_ss)
+
+rownames(treatment_tab$crm) <- c("Patients Treated by Model", "True Toxicity Probabilities")
+
+# i) accuracy
+
+dist_accuracy$crm <- recommended_dose(sims$crm)
+mean_accuracy$crm <- length(subset(dist_accuracy$crm,dist_accuracy$crm == best_dose_level)) / n_sims$crm
+#hist(dist_accuracy$crm,breaks=10)
+
+# ii) risk of overdosing
+
+dist_overdose$crm <- num_tox(sims$crm)
+mean_overdose$crm <- mean(dist_overdose$crm)
+#hist(dist_overdose$crm,breaks=10)
+
+# iii) trial length
+
+dist_length$crm <- sims$crm %>% escalation::trial_duration()
+mean_length$crm <- mean(dist_length$crm)
+#hist(dist_length$crm,breaks=10)
+
```

Comment:
This part of the code needs simplification. Many of them are almost the same except calling `crm` instead of `tpt`. Could you think of a clever way to make it future-proof in the next iteration?

Reply:
That was my #1 question for today - remind me if I forget to ask!